.github/workflows/fetch-merged-prs.yml

```
@@ -0,0 +1,55 @@
+name: Fetch Merged PRs After Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      release:
+        description: 'Release after which to fetch merged PRs'
+        required: true
+      branch:
+        description: 'Branch to fetch merged PRs from'
+        required: true
+        default: 'main'
+
+jobs:
+  fetch-merged-prs:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Get Release Date
+        id: get-release-date
+        run: |
+          RELEASE_DATE=$(curl -H "Accept: application/vnd.github.v3+json" -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" "https://api.github.com/repos/${{ github.repository }}/releases/tags/${{ github.event.inputs.release }}" | jq -r '.published_at')
+          echo "RELEASE_DATE=$RELEASE_DATE" >> $GITHUB_ENV
+
+      - name: Fetch merged PRs after release
+        id: fetch-prs
+        run: |
+          echo "PR Title,PR Link,PR Owner,Jira Task URL" > merged_prs.csv
+          pr_data=$(curl -H "Accept: application/vnd.github.v3+json" -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" "https://api.github.com/repos/${{ github.repository }}/pulls?state=closed&base=${{ github.event.inputs.branch }}&per_page=100")
```

```suggestion
          pr_data=$(curl -H "Accept: application/vnd.github.v3+json" -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" "https://api.github.com/repos/${{ github.repository }}/pulls?state=closed&base=${{ github.event.inputs.branch }}&per_page=100")
```

1. Does it also return closed-unmerged PRs?
2. We need only the top 100?

---

1. This call will fetch all PRs that are closed, including merged and unmerged.
In the code a bit lower I have a check for `.merged_at > $RELEASE_DATE`. For unmerged PRs `merged_at` will not be present, so in general it will remove all unmerged. To be sure it doesn't fail with an error, also added `merged_at != null` now.

2. No, but more than 100 will have to work with pagination, making the code more complex. I don't believe there's any way we can reach more than a hundred for the actual use-cases of the action.

---

Added pagination.